### DISCUSSION TITLE
Obtain the Hessian of a model?

### DISCUSSION THREAD

**Discussion by ghost** [SMTorg/smt#636] [Ideas]
Hi, I'm hoping to use this toolbox but I need the Hessian of the model. Is there an easy way to obtain this? 
Incidentally, I asked @hwangjt this a few years ago and I think he mentioned RMTSs but I've moved organizations since then and no longer have that email. 

As a bonus (not sure of the details of the techniques), if there is a convenient way to obtain the eigen decomposition of the Hessian that would be worth a gold star!

Thanks, Mike

**Comment by Paul-Saves** [comment]
- It depends on the model. For Kriging, you have only the derivative w.r.t to spatial variables but hessian could be implemented.  Similary, 
RMTS being splines, the Hessian is computed when training the model but no function yet allow the user to predict the Hessian at a  given point but it could be implemented. More here: https://smt.readthedocs.io/en/latest/_src_docs/surrogate_models.html 
- The eigen decomposition is easily obtained with numpy: np.linalg.eig(H)

**Reply by ghost** [reply]
Thanks. I want to avoid the extra burden of eig if I can avoid it so I'm wondering if any of the models exploit it. 

I saw the Hessian being calculated during training - any tips on how to export it at a given point? (Perhaps I should create a feature request?)
